### PR TITLE
Reword 'allows to' sentences

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
I noticed this in PR #6336

"This permission allows to list all spaces."
"This permission allows to create public links."

See https://english.stackexchange.com/questions/85069/is-the-construction-it-allows-to-proper-english

This sentence construction "allows to verb-infinitive" does not read nicely for my brain in English. It seems that it results from a "direct" translation of this kind of sentence in German (and other European languages).

I have replaced them with "allows using" "allows creating" etc.

## How Has This Been Tested?
N/A

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: ocis-pkg/service/grpc/client.go [hunk 1]
```diff
@@ -28,14 +28,14 @@ type ClientOptions struct {
 // Option is used to pass client options
 type ClientOption func(opts *ClientOptions)
 
-// WithTLSMode allows to set the TLSMode option for grpc clients
+// WithTLSMode allows setting the TLSMode option for grpc clients
```
Comment (phil-davis) [inline ocis-pkg/service/grpc/client.go:31]: Note: these kind of sentences can also be reworded to put the "to set" at the end, in the passive:

`WithTLSMode allows the TLSMode option for grpc clients to be set`

Either read OK to me, so maybe we just do the simple change that I have done here - "to set" => "setting"

File: services/eventhistory/pkg/service/service_test.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ var _ = Describe("EventHistoryService", func() {
 		close(bus)
 	})
 
-	It("Records events, stores them and allows to retrieve them", func() {
+	It("Records events, stores them and allows them to be retrieved", func() {
```
Comment (phil-davis) [inline services/eventhistory/pkg/service/service_test.go:45]: I turned the words around a bit for this one.

**Comment by phil-davis** [comment]
Do I need to write a changelog for this sort of thing?

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiAsyncUpload-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/22544/55/1) failed. Further test are cancelled...

**Review by EParzefall — APPROVED** [review]
Yes, typical German constructions. Great catches. Thanks, Phil!

**Pull request merged**